src/main/scala/interfaces/Converter.scala

```diff
@@ -0,0 +1,318 @@
+        val entry : ModelEntry = model.entries(fname)
+        entry match {
+            case SingleEntry(s) => return s 
+            case MapEntry(m: Map[Seq[String], String], els:String) =>
+                val filtered = m.filter(x => snapToOneLine(x._1.mkString(" ")) == args)
+                if (filtered.size >= 1) {
+                    return filtered.head._2.toString
+                }
+                return els
+        }
+    }
+
+
+    def translateSort(s:Sort) : String = {
+        s match {
+            case sorts.Set(els:Sort) => return "Set<" + translateSort(els)+">"
+            case sorts.Ref => return "$Ref"
+            case sorts.Snap => return "$Snap"
+            case sorts.Perm => return "$Perm"
+            case sorts.Seq(els:Sort) => return "Seq<" + translateSort(els)+">"
+            case _ => s.toString
+        }
+    }
+
+    def sortToType(s:Sort) : Option[ast.Type] = {
+        s match {
+            case sorts.Ref => return Some(ast.Ref)
+            case sorts.Int => return Some(ast.Int)
+            case sorts.Bool => return Some(ast.Bool)
+            case sorts.Perm => return Some(ast.Perm)
+            case _ => println("This type of Heap Field can not be handled : " + s.toString)
+                      return None
+        }
+    }
+    
+
+    def evaluateTerm(term:Term, model: Model) : String = {
+        term match {
+            case Unit => return "$Snap.unit"
+            case IntLiteral(n) => return term.toString
+            case t:BooleanLiteral => return t.value.toString
+            case Null() => return model.entries("$Ref.null").toString
+            case Var(id, sort) => {
+                val key = term.toString
+                //this can fail : TODO throw and catch exception
+                return model.entries(key).toString
+            }
+            case t@App(app, args) => {
+                println("Found type APP")
+                /* not tested yet, not sure for which examples this occurs on heap*/
+                var fname = app.id + "%limited"
+                if (!model.entries.contains(fname)){
+                    fname = app.id.toString
+                    if (!model.entries.contains(fname)){
+                        fname = fname.replace("[","<").replace("]",">")
+                    }
+                }
+                var arg = snapToOneLine(args.map(x => evaluateTerm(x, model)).mkString(" "))
+                return getFunctionValue(model, fname, arg)
+            }
+            case Combine(p0, p1) => {
+                val p0eval = evaluateTerm(p0, model)
+                val p1eval = evaluateTerm(p1, model)
+                return "($Snap.combine " + p0eval + " " + p1eval + ")"
+            }
+            case First(p) => {
+                val sub = evaluateTerm(p, model)
+                if (sub.startsWith("($Snap.combine")){
+                    return getParts(sub)(1)
+                }
```

Comment:
I guess this is still inherited from Nagini, but it would be much better if there was no stringification followed by parsing. Should `evaluateTerm` maybe return `ExtractedModelEntry` values? Maybe it doesn't yet have all the variants it needs to have?

Reply:
The problem is that the values stored in the model are only available as Strings, so those have to be parsed anyways and it was easier to just do the same with the values obtained from the heap. 
But it should still be possible to do the parsing of model values right in the evaluateTerm function, their types should be available. I will look into this again.

Reply:
FYI, @dohrau changed the model parser yesterday (https://github.com/viperproject/silver/commit/46f31079da934bd884f1d5aaf63d0c378d2ab2c6) so that values are no longer just strings but instances of a new ValueEntry class (that contain strings). Entries that contain function applications, like the ``$Snap.combine`` ones, are now parsed as instances of the ApplicationEntry class.

Reply:
@marcoeilers @dohrau I adjusted my code to these new entries but while debugging I noticed that negative integers are stored in an ApplicationEntry with name "-" and the positive integer as an argument. Is this actually intended?

Reply:
Yes. This is due to the [smt-lib](http://smtlib.cs.uiowa.edu/papers/smt-lib-reference-v2.6-r2017-07-18.pdf) representing negative numbers as "(- n)". The application entry simply reflects this structure as the parser does not perform any transformations or simplifications.

If you want to have integer values, you neet to convert the string values to integers at some point anyway. I suggest you also handle the negation there.